File: packages/witness/src/templates/cucumber-capybara/testivai_witness.rb
# TestivAI Visual Regression Helper
# Docs: https://testiv.ai/docs/frameworks/ruby/
#
# Requires the gem:  gem "testivai", group: :test
#
# This wrapper is optional — `Testivai.witness(page, name)` works directly.
# It exists so specs can call a bare `witness("name")`.
require "testivai"

module TestivaiWitness
  def witness(name, **options)
    Testivai.witness(page, name, **options)
  end
end

File: packages/witness/src/templates/selenium-testng/TestivAIWitness.java
// TestivAI Visual Regression Helper
// Docs: https://testiv.ai/docs#java
package testivai;

import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.WebDriver;

public class TestivAIWitness {
    public static Object witness(WebDriver driver, String name) {
        return ((JavascriptExecutor) driver).executeScript(
            "return window.testivaiWitness(arguments[0])", name
        );
    }
}
